src/client/sgidLogin/index.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { useEffect } from 'react'
+import i18next from 'i18next'
+import {
+  Button,
+  Hidden,
+  Link,
+  Typography,
+  createStyles,
+  makeStyles,
+} from '@material-ui/core'
+import GoLogo from '@assets/go-logo-graphics/go-main-logo.svg'
+import LoginGraphics from '@assets/login-page-graphics/login-page-graphics.svg'
+import { useDispatch } from 'react-redux'
+import rootActions from '../app/components/pages/RootPage/actions'
+
+import assetVariant from '../../shared/util/asset-variant'
+import sgidLoginActions from './actions'
+
+import { htmlSanitizer } from '../app/util/format'
+import Section from '../app/components/Section'
+import BaseLayout from '../app/components/BaseLayout'
+
+const URL_PREFIX_LENGTH = 11
+
+const useStyles = makeStyles((theme) =>
+  createStyles({
+    container: {
+      display: 'flex',
+      flexGrow: 1,
+      '-ms-flex': '1 1 auto',
+    },
+    loginContainer: {
+      display: 'flex',
+      width: '100%',
+      [theme.breakpoints.up('lg')]: {
+        width: '50%',
+      },
+    },
+    verticalAlign: {
+      display: 'flex',
+      width: '100%',
+      [theme.breakpoints.up('lg')]: {
+        alignItems: 'center',
+      },
+    },
+    signInButton: {
+      width: '250px',
+      minWidth: '120px',
+      marginRight: theme.spacing(2),
+    },
+    loginWrapper: {
+      display: 'block',
+      [theme.breakpoints.up('lg')]: {
+        // Gives the contents slightly more than enough height,
+        // so that validation messages do not shift the centering.
+        height: '400px',
+        maxHeight: '80vh',
+      },
+    },
+    headerGroup: {
+      marginBottom: theme.spacing(4),
+    },
+    logo: {
+      maxWidth: '130px',
+      width: '40%',
+    },
+    loginHeader: {
+      marginTop: theme.spacing(1),
+    },
+    loginReferral: {
+      fontSize: '0.85rem',
+      color: '#767676',
+      marginBottom: theme.spacing(4),
+    },
+    textInputGroup: {
+      marginBottom: theme.spacing(4),
+    },
+    graphicColorFill: {
+      backgroundColor: theme.palette.primary.dark,
+      width: '50vw',
+      height: '100%',
+      // allocates space for the government masthead when on gov variant
+      maxHeight: assetVariant === 'gov' ? 'calc(100vh - 28px)' : '100vh',
+      textAlign: 'center',
+      overflow: 'hidden',
+    },
+    loginGraphic: {
+      userDrag: 'none',
+      height: '100%',
+    },
+    '@media screen\\0': {
+      // Styles for Internet Explorer compatibility
+      logoLink: {
+        marginBottom: '0',
+      },
+    },
+    secondaryButton: {
+      fontWeight: 400,
+      marginLeft: theme.spacing(2),
+    },
+    resendOTPBtn: {
+      marginRight: 'auto',
+      '&:disabled': {
+        opacity: 0.5,
+      },
+    },
+  }),
+)
+
+const SgidLoginPage = (): JSX.Element => {
+  const dispatch = useDispatch()
+  const classes = useStyles()
+  const queryParams = new URLSearchParams(
```

Comment:
may i know why you decided to pass it via the hash instead of using query params? personally it took me a while to understand how this extracted the query params

on that note - it might help to defined 
`const URL_PREFIX_LENGTH = "#/ogp-login".length `
as it might be clearer as 11 is a magic number

Reply:
For some reason the query params were not being extracted properly, hence had to go with this approach.

Reply:
hmm still didn't work :/